### ISSUE TITLE
Query ArrayField more specificly

### ISSUE THREAD

**Issue by Vichoko** [nesdis/djongo#309]
#### One line description of the issue
Current query system with dictionaries it's too broad when you wan't to find all the looked up fields at the same time in the array field element.

#### Python script
Taken from the [querying array fields](https://djongo.readthedocs.io/docs/using-django-with-mongodb-array-field/#querying-array-fields)

**models.py**
```python
class Author(models.Model):
    name = models.CharField(max_length=200)
    email = models.EmailField()

    class Meta:
        abstract = True

    def __str__(self):
        return ''{}; {}".format(self.name, self.email)

class Entry(models.Model):


    headline = models.CharField(max_length=255)
    body_text = models.TextField()

    authors = models.ArrayModelField(
        model_container=Author,
    )

```
To explain the issue, picture this:
We have this authors in the database:
```python
[{'name': 'paul',
'name':'paul@email.com'},
{'name': 'paul2',
'name':'paul@email.com'}
```

Now, given an entry, you want to find all the the authors whichs have:
```name = 'paul'``` and ```email =  'paul@email.com'```, at the same time.

In this case, i want to match the first document in the db.
If i use the provided method

```python
> Entry.objects.filter(authors={'name': 'paul', 'email':'paul@email.com'})
<Author paul; paul@email.com>, <Author paul2; paul@email.com>
# i expect only <Author paul; paul@email.com>
```

So in other words, i want to enforce that all looked up fields appears at the same instance of the array field, it's a kind of AND instead of OR that the filter does.

**Comment by Vichoko** [comment]
I managed to do it more efficiently with DjongoManager and low-level raw queries with Entry.objects.mongo_aggregate i get the _id of every Entry i want, and then i gather the instances with a filter query, but i feel like i'm doing 2 queries here and i can do it more efficiently.
Any ideas?
